src/main/java/de/tum/in/www1/artemis/web/rest/TextAssessmentEventResource.java

```diff
@@ -0,0 +1,90 @@
+import org.springframework.web.bind.annotation.*;
+
+import de.tum.in.www1.artemis.domain.Course;
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.domain.analytics.TextAssessmentEvent;
+import de.tum.in.www1.artemis.repository.CourseRepository;
+import de.tum.in.www1.artemis.repository.TextAssessmentEventRepository;
+import de.tum.in.www1.artemis.repository.UserRepository;
+import de.tum.in.www1.artemis.service.AuthorizationCheckService;
+import de.tum.in.www1.artemis.web.rest.errors.EntityNotFoundException;
+
+/**
+ * REST controller for managing TextAssessmentEventResource.
+ */
+@RestController
+@RequestMapping("/api/text-assessment-event")
+public class TextAssessmentEventResource {
+
+    private final Logger log = LoggerFactory.getLogger(TextAssessmentEventResource.class);
+
+    private final TextAssessmentEventRepository textAssessmentEventRepository;
+
+    private final AuthorizationCheckService authCheckService;
+
+    private final UserRepository userRepository;
+
+    private final CourseRepository courseRepository;
+
+    public TextAssessmentEventResource(TextAssessmentEventRepository textAssessmentEventRepository, AuthorizationCheckService authCheckService, UserRepository userRepository,
+            CourseRepository courseRepository) {
+        this.textAssessmentEventRepository = textAssessmentEventRepository;
+        this.authCheckService = authCheckService;
+        this.userRepository = userRepository;
+        this.courseRepository = courseRepository;
+    }
+
+    /**
+     * This function retrieves all of the events from the 'text_assessment_event' table
+     * @return returns a List of TextAssessmentEvent's
+     */
+    @GetMapping("/get-events")
+    @PreAuthorize("hasRole('INSTRUCTOR')")
+    public ResponseEntity<List<TextAssessmentEvent>> findAllEvents() {
+        List<TextAssessmentEvent> events = textAssessmentEventRepository.findAll();
+        return ResponseEntity.ok().body(events);
+    }
+
+    /**
+     * This function adds an assessment event into the text_assessment_event table.
+     * @param event to be added
+     * @return the status of the finished request
+     * @throws URISyntaxException
+     */
+    @PostMapping("/add-event")
+    @PreAuthorize("hasRole('TA')")
+    public ResponseEntity<Void> addAssessmentEvent(@RequestBody TextAssessmentEvent event) throws URISyntaxException {
+        log.debug("REST request to save assessmentEvent : {}", event);
+
+        // A new assessmentEvent cannot already have an ID
+        if (event.getId() != null) {
+            return ResponseEntity.badRequest().build();
+        }
+
+        // avoid access from tutor if they are not part of the course
+        User user = userRepository.getUserWithGroupsAndAuthorities();
+        try {
+            Course course = courseRepository.findByIdElseThrow(event.getCourseId());
+            if (!authCheckService.isAtLeastTeachingAssistantInCourse(course, user)) {
```

Comment:
you can use this check in here: `authCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.X, course, user)`

Reply:
I [changed ](https://github.com/ls1intum/Artemis/pull/3704/commits/2527556401de63c7f80f5236a301419e7c1bc63c) to this, but I can see it had the respective test [failing](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS2208-JAVATEST-19) since it just throws. I [reverted](https://github.com/ls1intum/Artemis/pull/3704/commits/7fb5eec926de8f3498bc6d070a17c7563f087b39) to the previous one, as it looks cleaner than catching multiple Exception instead.

Reply:
hmm, okay. I suggested this because afair, we want to get rid of this pattern for new endpoints. It is suggested in the best practices part of [documentation](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/server/#rest-endpoint-best-practices-for-authorization). If it's a problem that can't be solved very easily, that's fine by me.